nautobot/ipam/models.py

```diff
@@ -1122,8 +1122,8 @@ def clean(self):
 
         closest_parent = self._get_closest_parent()
         # Validate `parent` can be used as the parent for this ipaddress
-        if self.parent and closest_parent:
-            if self.parent != closest_parent:
+        if closest_parent is not None:
+            if self.parent is not None and self.parent != closest_parent:
```

Comment:
Why this change here?

Reply:
Merging in the case formerly at lines 1150-1154 below - if self.parent is None but we can find a closest_parent, then we need to update self.parent = closest_parent before continuing (lines 1135-1136).